tesseract-core/src/query.rs

```diff
@@ -490,3 +504,22 @@ impl FromStr for RateQuery {
         })
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_validate() {
+        let mut q = Query::new();
+        assert!(q.validate().is_ok());
```

Comment:
@jspeis is this what you mean? General question too, should I be putting multiple assertions in one function or should I do one assertion per function, which one is better practice in Rust?

Reply:
@AnthonyHewins, something like that -- yes is what I had in mind with two minor notes to the example above exception: to be a valid query a query should have at least one drilldown and at least one measure (cc @hwchen) so technically I would expect the default query which has a empty vec for both drilldowns and measures to be _invalid_. 

In terms of which is a better practice, I think it's more a matter of style and preference. Overall I think it's good to have a test function focus on one task, which tends to make it easier to give a function a descriptive name that others will be able to follow and understand. In this particular case, I could see it going either way because the logic is not particularly complicated.

Reply:
It is a little flexible, but I generally have more than one assertion in a test function. It really depends on the test.

For this series of tests, I might do:
- validate_missing_params (which could include assertions for missing drilldown, missing measure, and both)
- validate_rca
- etc.
Let me know if this clarifies things.

Reply:
And yes, a valid query should have both a drilldown and a measure.